pkg/reconciler/pipeline/watcher.go

```diff
@@ -0,0 +1,135 @@
+package pipeline
+
+import (
+	"context"
+	"fmt"
+
+	appv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+)
+
+func setResourceStatus(c client.Client, pipeline *dfv1.Pipeline) error {
+	// get the daemon deployment and update the status of it to the pipeline
+	var daemonDeployment appv1.Deployment
+	if err := c.Get(context.TODO(), client.ObjectKey{
+		Namespace: pipeline.GetNamespace(),
+		Name:      pipeline.GetDaemonDeploymentName(),
+	}, &daemonDeployment); err != nil {
+		return err
+	}
+	if msg, serviceType, reason, status := getDeploymentStatus(&daemonDeployment); status {
+		pipeline.Status.MarkServiceHealthy(serviceType, reason, msg)
+	} else {
+		pipeline.Status.MarkServiceNotHealthy(serviceType, reason, msg)
```

Comment:
If it's not healthy, return and requeue.

Reply:
Just thinking, do we need to requeue in case of not healthy, as this fn will be going to triggered everytime if there is any change in status of daemon deployment due to `predicate.ResourceVersionChangedPredicate{}` ?

Reply:
Makes sense.